### ISSUE TITLE
Resume model training from checkpoint

### ISSUE THREAD

**Issue by schrum2** [schrum2/MarioDiffusion#83]
Currently, if you try to train a model and a directory already exists, the attempt simply fails.

However, sometimes we will start to train a model, and something will cause the code to crash, or the computer might lose power, etc. In such circumstances, it might be desirable to resume training from where it left off. This won't be possible, but we should at least be able to resume training from the last saved checkpoint.

So, here is how the code for train_diffusion should behave moving forward.
1) Check if directory exists.
2) If directory exists, then prompt asks user if they want to resume training from last saved checkpoint
3) If yes, then that checkpoint is loaded and training continues.

Note that we may need to save some additional information during training besides the checkpoints to make this possible. We can infer which epoch the training was on from the checkpoint directory name, but we need information about the learning rate schedule and possibly other information too. If you develop this feature with AI, it should be able to help you find these aspects of the training state that are needed.

Resuming in this manner will also resume writing to the log files, but there could be some minor issues here. For example, if training crashed at epoch 33 and the last saved checkpoint was epoch 20, then there are 13 lines of training loss logging in the log file that are now irrelevant and will be replaced by the reloaded simulation. Similarly, the caption match and validation loss are only logged every 5 epochs. These issues need to be taken into account.
